repository/maven.go

```diff
@@ -1,24 +1,27 @@
 package repository
 
 import (
+	"os"
+	pathlib "path"
+
 	"github.com/clbanning/mxj"
 	liberr "github.com/jortel/go-utils/error"
 	"github.com/konveyor/tackle2-addon/command"
 	"github.com/konveyor/tackle2-hub/api"
 	"github.com/konveyor/tackle2-hub/nas"
-	"os"
-	pathlib "path"
 )
 
-//
 // Maven repository.
 type Maven struct {
 	Remote
 	BinDir string
 	M2Dir  string
 }
 
-//
+func (r *Maven) CreateSettingsFile() (string, error) {
+	return r.writeSettings()
```

Comment:
Let's move the content of writeSettings() here.  Let's change the signature to pass in the path for better utility.
CreateSettingsFile(path string) (err error)
With the following flow:

var settings string

1. Get the identity.
if found, settings = id.Settings else settings = _EmptySettings_
2. m, err = parse XML
3.err =  r.injectCacheDir(m *mxj.Map)
4. err = r.injectProxies(m *mxj.Map)
5. write file.

Reply:
EmptySettings would a _const_ with something like:
```
<settings xmlns="http://maven.apache.org/SETTINGS/1.2.0" 
                 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
                 xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.2.0
                 http://maven.apache.org/xsd/settings-1.2.0.xsd">
</settings>
```

Reply:
updated